### DISCUSSION TITLE
using oclAsType with inheritance and overriding

### DISCUSSION THREAD

**Discussion by phillipabennett** [useocl/use#68] [Q&A]
According to the OMG specification for OCL (section 7.5.8. for OCL 2.4),

> whenever properties are redefined within a type, the property of the supertypes can be accessed using the oclAsType() operation. Whenever we have a class B as a subtype of class A, and a property p1 of both A and B, we can write:
> context B inv:
>     self.oclAsType(A).p1 -- accesses the p1 property defined in A 
>     self.p1 -- accesses the p1 property defined in B

In USE, say we have:

    class A operations op() end
    class B < A operations op() end

In class B, using 
    self.oclAsType(A).op() 
gives the following error 
    _Target type `A' is not a subtype of the source expression's type `B'_

Is is that the USE tool does not implement calling an overridden operation in a superclass?
How can I get access to the operation in the superclass?

Thank you!

**Comment by h-man2** [comment]
Unfortunately, the behaviour described in the OCL specification is not supported. I'm going to open an issue for this.

I tried using a let-expression, but `let myB:A = B.allInstances()->any(true) in myB.op()` does not call the "super" implementation. So, for now there is no way access these properties.
